Pipeline/PackageBuildOutputs/ArtifactoryHelpers.groovy

```diff
@@ -0,0 +1,182 @@
+	if ( verbose ) println( "** Request: " + httpPut );
+	
+	FileEntity fileEntity = new FileEntity(new File(fileName));
+	fileEntity.setContentType("binary/octet-stream");
+	httpPut.setEntity(fileEntity);
+
+	HttpClientContext clientContext = HttpClientContext.create();
+
+	CredentialsProvider provider = new BasicCredentialsProvider();
+	UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(user, password);
+	provider.setCredentials(AuthScope.ANY, credentials);
+	clientContext.setCredentialsProvider(provider);
+
+	PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+	connectionManager.setMaxTotal(conPoolSize);
+	connectionManager.setDefaultMaxPerRoute(conPoolSize);
+
+	HttpClientBuilder builder = HttpClientBuilder.create().setConnectionManager(connectionManager).setDefaultRequestConfig(requestConfig).useSystemProperties()
+	builder.setRetryHandler(new CustomRetryHandler(retryCount, verbose));
+
+
+	CloseableHttpClient httpClient = builder.build()
+	
+	HttpResponse response = httpClient.execute(httpPut, clientContext);
+	
+	if ( verbose ) println( "** Response: " + response );
+		
+	int statusCode = response.getStatusLine().getStatusCode();
+
+	if ((statusCode != HttpStatus.SC_CREATED) && (statusCode != HttpStatus.SC_OK)) {
+		if (response.getEntity() != null) {
+			InputStream source = response.getEntity().getContent();
+			byte[] buffer = new byte[8192];
+			int read;
+			while ((read = source.read(buffer)) != -1) {
+				System.err.write(buffer, 0, read);
+			}
+		}
+		throw new RuntimeException("Artifactory upload failed: " + statusCode);
+	}
+	httpClient.close();
+}
+
+def download(String url, String fileName, String user, String password, boolean verbose) throws ClientProtocolException, IOException  {
+	RequestConfig requestConfig = RequestConfig
+		.custom()
+		.setSocketTimeout(timeoutMilliSeconds)
+		.setConnectTimeout(timeoutMilliSeconds)
+		.setCircularRedirectsAllowed(true)
+		.setExpectContinueEnabled(true)
+		.build();
+
+	HttpGet get = new HttpGet(url);
+	get.setConfig(requestConfig);
+
+	HttpClientContext clientContext = HttpClientContext.create();
+	CredentialsProvider provider = new BasicCredentialsProvider();
+	UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(user, password);
+	provider.setCredentials(AuthScope.ANY, credentials);
+	clientContext.setCredentialsProvider(provider);
+
+	CloseableHttpClient httpClient = HttpClients.custom().useSystemProperties().build()
+	
+	HttpResponse response = httpClient.execute(get, clientContext);
+	InputStream source = response.getEntity().getContent();
+	FileOutputStream out = new FileOutputStream(fileName);
+	byte[] buffer = new byte[8192];
+	int read;
+	while ((read = source.read(buffer)) != -1) {
+		out.write(buffer, 0, read);
+	}
+	out.close();
+	httpClient.close();
+}
+
+
+//Parsing the command line
+def run(String[] cliArgs)
+{
+	System.setProperty("com.ibm.jsse2.overrideDefaultTLS", "true")
```

Comment:
Can you remove this line? 
This is something that should be pass in JAVA Opts (if needed) -Dcom.ibm.jsse2.overrideDefaultTLS=true. In OCP demos I hard-coded it in the groovy script (to not modify Ansible playbooks)..